extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/HttpSessionContext.java

```diff
@@ -65,12 +65,19 @@ public boolean isActive() {
     }
 
     @Override
-    public Collection<ContextInstanceHandle<?>> getAll() {
-        HttpServletRequest httpServletRequest = servletRequest();
-        if (httpServletRequest != null) {
-            return new ArrayList<>(getContextualInstances(httpServletRequest).getPresentValues());
-        }
-        return Collections.emptyList();
+    public ContextState getState() {
+        return new ContextState() {
+
+            @Override
+            public Map<InjectableBean<?>, Object> getContextualInstances() {
+                HttpServletRequest httpServletRequest = servletRequest();
```

Comment:
So ATM this is not a `ManagedContext` which means we can't restore it, with a `ThreadContextProvider`? But we don't need to because it gets its values from the servlet request, which _is_ restored by CP. That's lucky because we don't support `ThreadContextProvider`s which depend on one another.

Reply:
Yes, it would be more complicated to make this a managed context. Honestly, this context is not supposed to be used in production ;-).

Reply:
No, let's not change it because it should just work currently, the way it's implemented :)

Reply:
This isn't really for propagation, it's just a quick grab for all instance in that context.
But yes, it should incidentally be kept up to date.